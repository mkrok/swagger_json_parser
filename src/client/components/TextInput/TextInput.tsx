import * as React from 'react';
import styled from 'styled-components';

const TextInputStyled = styled.input`
  font-size: 1em;
  width: 20em;
  height: 2em;
  background-color: #bbb;
`;

const LabelStyled = styled.label`
  font-size: 1em;
  color: #ccc;
  margin-right: 0.5em;
`;

// type TextInputProps = {
//   disabled?: boolean;
//   label?: string;
//   value: string;
//   onChange?: React.ChangeEventHandler;
// };

const TextInput = ({ label = '', onChange = null, value }) => (
  <>
    <LabelStyled>{label}</LabelStyled>
    <TextInputStyled onChange={onChange} type="text" value={value} />
  </>
);

export default TextInput;
