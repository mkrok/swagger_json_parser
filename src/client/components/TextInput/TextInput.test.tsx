import * as React from 'react';
import renderer from 'react-test-renderer';
import TextInput from './TextInput';

describe('TextInput', () => {
  it('should match snapshot when all props are defined', () => {
    const value = '';
    const props = {
      disabled: false,
      label: 'Enter your name: ',
      value,
      onChange: () => {},
    };
    const component = renderer.create(<TextInput {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });
  it('should match snapshot when only necessary props are defined', () => {
    const value = '';
    const props = {
      value,
    };
    const component = renderer.create(<TextInput {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
