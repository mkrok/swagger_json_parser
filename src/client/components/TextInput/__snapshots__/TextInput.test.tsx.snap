// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should match snapshot when all props are defined 1`] = `
Array [
  <label
    className="sc-gueYoa ggGRkb"
  >
    Enter your name: 
  </label>,
  <input
    className="sc-beqWaB hoAMot"
    onChange={[Function]}
    type="text"
    value=""
  />,
]
`;

exports[`TextInput should match snapshot when only necessary props are defined 1`] = `
Array [
  <label
    className="sc-gueYoa ggGRkb"
  />,
  <input
    className="sc-beqWaB hoAMot"
    onChange={null}
    type="text"
    value=""
  />,
]
`;
